cmd/devicesimulator/main.go

```diff
@@ -167,23 +165,35 @@ func reportVersion(versionFormat *string) error {
 
 func startAgent(ctx context.Context, agent *agent.Agent, log *logrus.Logger, agentInstance int) {
 	activeAgents.Inc()
-	prefix := agent.GetLogPrefix()
-	err := agent.Run(ctx)
-	if err != nil {
-		// agent timeout waiting for enrollment approval
-		if errors.Is(err, wait.ErrWaitTimeout) {
-			log.Errorf("%s: agent timed out: %v", prefix, err)
-		} else if ctx.Err() != nil {
-			// normal teardown
-			log.Infof("%s: agent stopped due to context cancellation.", prefix)
-		} else {
-			log.Fatalf("%s: %v", prefix, err)
+	err := wait.PollImmediateWithContext(ctx, 2*time.Second, 5*time.Minute, func(ctx context.Context) (bool, error) {
+		// timeout after 30s and retry
```

Comment:
why?

Reply:
The above code tries every 2 seconds for 5 minutes. This 30 sec timeout is for each attempt, so an attempt will be cancelled if it's got stuck for any reason.
BTW this and code is copy/paste from `approveAgent()`  func, i.e. all numbers are from there.

Reply:
Moved all those numbers to constants.